Makefile

```diff
@@ -0,0 +1,47 @@
+BINDIR=/usr/local/bin
+UNITDIR=/etc/systemd/system
+SYSCONFDIR=/usr/local/etc
+DEFAULTDIR=$(SYSCONFDIR)/default
+DESTDIR=
+PROGNAME=alertmanager_matrix
+
+all: bin/$(PROGNAME) $(PROGNAME).service
+
+ROOT_DIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+
+.PHONY: clean dist rpm srpm install-unit install-prog install-default install
+
+clean:
+	cd $(ROOT_DIR) || exit $$? ; find -name '*.pyc' -o -name '*~' -print0 | xargs -0 rm -f
+	cd $(ROOT_DIR) || exit $$? ; rm -rf *.tar.gz *.rpm
+	cd $(ROOT_DIR) || exit $$? ; rm -rf bin $(PROGNAME).service
+
+dist: clean
```

Comment:
Why does this depend on `clean`?

Reply:
That's standard `autotools` and GNU practice.  You should not build a `dist` package with whatever junk might be in the source directory -- first you must `clean`.

Reply:
Technically you must `distclean` but we don't need the distinction between `clean` and `distclean` in a Go build.